solr/core/src/test/org/apache/solr/core/TestSolrConfigHandler.java

```diff
@@ -1011,4 +1011,61 @@ public static LinkedHashMapWriter getRespMap(String path, RestTestHarness restHa
       return new LinkedHashMapWriter<>();
     }
   }
+
+  public void testCacheDisableSolrConfig() throws Exception {
+    restTestHarness.setServerProvider(() -> getBaseUrl());
+    MapWriter confMap = getRespMap("/admin/metrics", restTestHarness);
+    assertNotNull(
+        confMap._get(
+            asList("metrics", "solr.core.collection1", "CACHE.searcher.fieldValueCache"), null));
+    // Here documentCache is disabled at initialization in SolrConfig
+    assertNull(
+        confMap._get(
+            asList("metrics", "solr.core.collection1", "CACHE.searcher.documentCache"), null));
+  }
+
+  public void testSetPropertyCacheSize() throws Exception{
+    RESTfulServerProvider oldProvider = restTestHarness.getServerProvider();
+    // Changing cache size
+    String payload = "{\n" + " 'set-property' : { 'query.documentCache.size': 399} \n" + " }";
+    restTestHarness.setServerProvider(oldProvider);
+    runConfigCommand(restTestHarness, "/config", payload);
```

Comment:
not sure I understand the behavior of the "oldProvider"?  Looks like we get it and then set it again?

Reply:
Actually in-between I changed it. So just reset it back to old behaviour. 
` restTestHarness.setServerProvider(() -> getBaseUrl());`